### PR TITLE
Merge step 2

### PR DESCRIPTION
**Pull request by alilee**
Step 2 by alilee

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+struct Moving {
+    dir: Direction,
+}
+
+fn list_snakes(query: Query<(&Location, &Moving), With<Snake>>) {
+    for (loc, moving) in query.iter() {
+        println!("Snake: {:?}, {:?}", loc, moving);
+    }
+}
+
```
Comment (interacsion) [inline src/main.rs]: Maybe this shouldn't be in main, but in different module(s)?

Comment (alilee) [reply src/main.rs]: Have moved stuff into sub-modules, and I know there is a way to encapsulate this more by making it a plugin.  Happy for someone else to add their ideas.

Comment (interacsion) [reply src/main.rs]: Looks way better in my opinion.

Comment (alilee) [reply src/main.rs]: Cool - want to add input or movement each tick?

Comment (interacsion) [reply src/main.rs]: Sure, I can try to

**Comment by interacsion** [comment]
see #3. it isn't ready to merge yet though.

**Review by interacsion — CHANGES_REQUESTED** [review]
We should make sure the player doesn't go backwards by changing the moving direction multiple times between ticks.

**Comment by alilee** [comment]
I am starting to like this. It is hugely over-engineered for this step, but I think it is going to make the next steps quite manageable. Just the backwards problem to go? I feel like it should be able to be solved inside `input`.

**Comment by alilee** [comment]
> We should make sure the player doesn't go backwards by changing the moving direction multiple times between ticks.

This actually doesn't matter so much because that move will collide with the first segment of the body. Maybe player should be protected from that, but could be done by using the body somehow. I think we should move on and solve that particular edge-case later?

**Comment by interacsion** [comment]
I am not sure. I think we should have it as a part of the input module.

**Comment by alilee** [comment]
> I am not sure. I think we should have it as a part of the input module.

Ok happy with that. Do you want to jam it back in and merge?

**Pull request merged**